src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBrokerHelper.cs

```diff
@@ -153,13 +160,21 @@ public async Task<string> GetBrokerAuthTokenSilentlyAsync(IDictionary<string, st
                 }
             }
 
-            _logger.Info("Android Broker AddAccount didn't return any results. ");
+            _logger.Info("Android Broker didn't return any results.");
             return null;
         }
 
         public void GetBrokerAccountInfo(IDictionary<string, string> brokerPayload, Activity callerActivity)
         {
-            var loginHint = GetValueFromBrokerPayload(brokerPayload, BrokerParameter.LoginHint);
+            string loginHint = GetValueFromBrokerPayload(brokerPayload, BrokerParameter.LoginHint);
+            if (string.IsNullOrEmpty(loginHint))
+            {
+                //While it is very unlikely that we end up with both the homeAccountID and the username equal to null
+                //We should still notify the user via log message if this were to occur.
+                _logger.Error(MsalErrorMessage.NoUPNOrAccountIDForSilentBrokerAuth);
```

Comment:
>MsalErrorMessage.NoUPNOrAccountIDForSilentBrokerAuth [](start = 30, length = 52)

about this error message, is loginHint actually the homeAccountID and/or UPN? because i thought we were getting those from the IAccount object and login hint was given by the developer, so should this error message actually be .NoLoginHintProvided?

Reply:
The login hint is considered the username for the broker so we can use either the loginhint or the UPN from the account object. So for the broker silent requests. we need to have either the UPN(which may come from the login hint) or the homeaccount ID

Reply:
removing check here